src/parse_logcfg.c

```diff
@@ -672,33 +683,43 @@ static int cfg_countrylist(const cfg_arg_t arg) {
     If the input was not a file name we directly copy it into
     country_list_raw (must not have a preceding contest name). */
 
-    g_strlcpy(buffer, parameter, sizeof(buffer));
+    buffer = (char *)calloc(buffer_len, sizeof(char));
+    g_strlcpy(buffer, parameter, buffer_len);
     g_strchomp(buffer);	/* drop trailing whitespace */
 
+    printf("%s\n", buffer);
     if ((fp = fopen(buffer, "r")) != NULL) {
-
 	char *prefix = g_strdup_printf("%s:", whichcontest);
 
-	while (fgets(buffer, sizeof(buffer), fp) != NULL) {
-
-	    g_strstrip(buffer);   /* no leading/trailing whitespace*/
+	while ((read = getline(&buffer, &buffer_len, fp)) != -1) {
+	    if (buffer_len > 0) {
+		if (errno == ENOMEM) {
+		    fprintf(stderr, "Error in: %s:%d\n", __FILE__, __LINE__);
+		    perror("RuntimeError: ");
+		    exit(EXIT_FAILURE);
+		}
+		g_strstrip(buffer);   /* no leading/trailing whitespace*/
 
-	    /* accept only a line starting with the contest name
-	     * (CONTEST=) followed by ':' */
-	    if (strncasecmp(buffer, prefix, strlen(prefix)) == 0) {
-		country_list_raw = buffer + strlen(prefix); // skip prefix
-		break;
+		/* accept only a line starting with the contest name
+		* (CONTEST=) followed by ':' */
+		if (strncasecmp(buffer, prefix, strlen(prefix)) == 0) {
+		    country_list_raw = buffer + strlen(prefix); // skip prefix
+		    break;
+		}
 	    }
 	}
 
 	g_free(prefix);
 	fclose(fp);
+
     } else {	/* not a file */
-	char *colon = index(buffer, ':');
-	if (colon != NULL) {
-	    country_list_raw = colon + 1;   // skip optional contest name
-	} else {
-	    country_list_raw = buffer;
+	if (buffer != NULL) {
```

Comment:
buffer is never NULL at this point. In case the previous calloc fails, then the following g_strlcpy (line 687) will already cause a segmentation violation.